lib/curlx/fopen.c

```diff
@@ -97,15 +97,17 @@ static bool fix_excessive_path(const TCHAR *in, TCHAR **out)
 
 #ifndef _UNICODE
   /* convert multibyte input to unicode */
-  needed = mbstowcs(NULL, in, 0);
-  if(needed == (size_t)-1 || needed >= max_path_len)
+  if(mbstowcs_s(&needed, NULL, 0, in, 0))
+    goto cleanup;
+  if(needed >= max_path_len)
     goto cleanup;
   ++needed; /* for NUL */
```

Comment:
```suggestion
```
the mbstowcs_s version includes the null terminator in needed wide characters, unlike mbstowcs